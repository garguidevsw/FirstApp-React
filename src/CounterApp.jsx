import { useState } from 'react';
import PropTypes from 'prop-types';

export const CounterApp = ({ value }) => {

    const [counter, setCounter] = useState(value);

    const addHandle = (event) => {
        setCounter( counter + 1 );
        // setCounter( (c) => c + 1 );
    }

    const subHandle = (event) => {
        setCounter( counter - 1 );
    }

    const resetHandle = (event) => {
        setCounter( value );
    }
    
  return (
    <>
        <h1>CounterApp</h1>
        <h2> { counter } </h2>

        <button onClick={ subHandle }>-1</button>
        <button onClick={ addHandle }>+1</button>
        <button onClick={ resetHandle }>Reset</button>
    </>
  )
}


CounterApp.propTypes = {
    value: PropTypes.number
}

CounterApp.defaultProps = {
    value: 0
}