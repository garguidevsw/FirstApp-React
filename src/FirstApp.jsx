
import PropTypes from 'prop-types';

export const FirstApp = ({ title, subtitle, number }) => {

  return (
    <>
        <h1 data-testid="test-title">{ title }</h1>
        <p>{ subtitle }</p>
        <p>{ subtitle }</p>
        <p>{ number + 2 }</p>
    </>
  )
}

FirstApp.propTypes = {
    title: PropTypes.string.isRequired,
    subtitle: PropTypes.string,
    number: PropTypes.number
}

FirstApp.defaultProps = {
    subtitle: 'Este es un subtitulo por defecto',
    number: 0
}
