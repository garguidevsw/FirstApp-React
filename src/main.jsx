import React from 'react';
import ReactDOM from 'react-dom/client';
import { FirstApp } from './FirstApp';
import { CounterApp } from './CounterApp';

import './styles.css';

ReactDOM.createRoot( document.getElementById('root')).render(
    <React.StrictMode>
        {/* <FirstApp title='Hola, soy Goku!' subtitle="Este es el subtitulo..." /> */}
        {/* <FirstApp title='Hola, soy Goku!' /> */}

        <CounterApp value={ 100 } />
    </React.StrictMode>
);