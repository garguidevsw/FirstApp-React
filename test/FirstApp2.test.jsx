import { render, screen } from "@testing-library/react"
import { FirstApp } from "../src/FirstApp"

describe('Pruebas en FirstApp2', () => {

    const title = 'Hola, soy Goku!';
    const subtitle = 'Soy un subtitulo';

    test('debe de hacer match con el snapshot', () => {
        const { container } = render( <FirstApp title={ title } /> );
        expect( container ).toMatchSnapshot();
    });

    test('debe mostrar el title Hola, soy Goku!', () => {
        
        render( <FirstApp title={ title } /> );
        expect( screen.getByText(title) ).toBeTruthy();
        // screen.debug();

    });

    test('debe mostrar el title en un h1', () => {
        
        render( <FirstApp title={ title } /> );
        expect( screen.getByRole( 'heading', { level: 1 } ).innerHTML ).toContain( title );

    });

    test('debe de mostrar el subtitle enviado por props', () => {

        render( <FirstApp title={ title } subtitle={ subtitle } /> );
        expect( screen.getAllByText(subtitle).length ).toBe(2);

    })
})