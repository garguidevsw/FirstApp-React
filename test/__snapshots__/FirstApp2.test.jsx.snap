// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en FirstApp2 debe de hacer match con el snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
    Hola, soy Goku!
  </h1>
  <p>
    Este es un subtitulo por defecto
  </p>
  <p>
    Este es un subtitulo por defecto
  </p>
  <p>
    2
  </p>
</div>
`;
